ci/kind/kind-upgrade.sh

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 Antrea Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eo pipefail
+LATEST_KIND_VERSION=$(curl -sSL https://api.github.com/repos/kubernetes-sigs/kind/releases/latest | jq -r '.tag_name' | awk '{print substr($1, 2)}')
```

Comment:
any concern that we may hit the Github API rate limit?

Reply:
@antoninbas I am considering if we can refer to the KIND_VERSION in kind.yml to align it with the version between GitHub Action and the static testbed. Additionally, if we want to quickly verify CI with an older version of kind, we can simply change this KIND_VERSION in PR and run tests on both GitHub Action and the static jenkins jobs.

Reply:
What we could do is maintain a file (for example, `ci/kind/version`) that stores the Kind version that we currently use for testing. Then we use the contents of this file everywhere (Github workflows, Jenkins scripts) to determine which version of Kind to use. IMO, it will be better than parsing `kind.yml` to retrieve that version. It will also avoid the duplicate `KIND_VERSION` definitions in `.github/workflows`:
```
$ git grep "KIND_VERSION: "
conformance.yml:  KIND_VERSION: v0.18.0
kind.yml:  KIND_VERSION: v0.20.0
netpol_cyclonus.yml:  KIND_VERSION: v0.20.0
```

Reply:
Thanks @XinShuYang  and @antoninbas , updated the code please check.